test/widgets/recent_dm_conversations_test.dart

```diff
@@ -206,39 +217,95 @@ void main() {
       });
 
       group('1:1', () {
-        testWidgets('has right title/avatar', (WidgetTester tester) async {
-          final user = eg.user(userId: 1);
-          final message = eg.dmMessage(from: eg.selfUser, to: [user]);
-          await setupPage(tester, users: [user], dmMessages: [message]);
+        group('has right avatar/title', () {
+          Future<void> checkRecipient(WidgetTester tester, {
+            required bool isBot,
+            required bool looksBot
+          }) async {
+            final user = eg.user(userId: 1, isBot: isBot);
+            final message = eg.dmMessage(from: eg.selfUser, to: [user]);
+            await setupPage(tester, users: [user], dmMessages: [message]);
+
+            checkAvatar(tester, DmNarrow.ofMessage(message, selfUserId: eg.selfUser.userId));
+            checkTitle(tester, user.fullName);
+            checkBotIcon(count: looksBot ? 1 : 0);
+          }
 
-          checkAvatar(tester, DmNarrow.ofMessage(message, selfUserId: eg.selfUser.userId));
-          checkTitle(tester, user.fullName);
+          testWidgets('bot recipient -> shows bot icon', (tester) async {
+            await checkRecipient(tester, isBot: true, looksBot: true);
+          });
+
+          testWidgets('non-bot recipient -> shows no bot icon', (tester) async {
+            await checkRecipient(tester, isBot: false, looksBot: false);
+          });
         });
 
-        testWidgets('no error when user somehow missing from store.users', (WidgetTester tester) async {
-          final user = eg.user(userId: 1);
-          final message = eg.dmMessage(from: eg.selfUser, to: [user]);
-          await setupPage(tester,
-            users: [], // exclude user
-            dmMessages: [message],
-          );
+        group('no error when user somehow missing from store.users', () {
+          Future<void> checkRecipient(WidgetTester tester, {
+            required bool isBot,
+            required bool looksBot
+          }) async {
+            final user = eg.user(userId: 1, isBot: isBot);
+            final message = eg.dmMessage(from: eg.selfUser, to: [user]);
+            await setupPage(tester,
+              users: [], // exclude user
+              dmMessages: [message],
+            );
+
+            checkAvatar(tester, DmNarrow.ofMessage(message, selfUserId: eg.selfUser.userId));
+            checkTitle(tester, '(unknown user)');
+            checkBotIcon(count: looksBot ? 1 : 0);
+          }
 
-          checkAvatar(tester, DmNarrow.ofMessage(message, selfUserId: eg.selfUser.userId));
-          checkTitle(tester, '(unknown user)');
+          testWidgets('bot recipient -> shows no bot icon', (tester) async {
+            await checkRecipient(tester, isBot: true, looksBot: false);
+          });
+
+          testWidgets('non-bot recipient -> shows no bot icon', (tester) async {
+            await checkRecipient(tester, isBot: false, looksBot: false);
+          });
```

Comment:
```dart
        group('no error when user somehow missing from store.users', () {
          // […]
          testWidgets('bot recipient -> shows no bot icon', (tester) async {
            await checkRecipient(tester, isBot: true, looksBot: false);
          });

          testWidgets('non-bot recipient -> shows no bot icon', (tester) async {
            await checkRecipient(tester, isBot: false, looksBot: false);
          });
```

The "bot recipient" test doesn't actually simulate a bot recipient, right, because these tests are checking what happens when the user data is missing. I see that the `User` passed to `eg.dmMessage` has `isBot: true`, but the `isBot` doesn't actually get written into the `DmMessage` object, so the app code can't treat this case differently than the "non-bot recipient" case.

Simplest I think to leave this test is it was, but with an added `checkBotIcon(count: 0);`.